homework/tai_nguyen/lesson7/695_max_area_of_island.ts

```diff
@@ -0,0 +1,44 @@
+/**
+ *
+ * @link https://leetcode.com/problems/max-area-of-island/
+ */
+
+function maxAreaOfIsland(grid: number[][]): number {
+  const m = grid.length;
+  const n = grid[0].length;
+  let count = 0,
+    res = 0;
+
+  function isInsideGrid(i: number, j: number) {
+    return i >= 0 && i < m && j >= 0 && j < n;
+  }
+
+  function dfs(row: number, col: number) {
+    count++;
+    grid[row][col] = 0;
+
+    const dr = [-1, 0, 1, 0];
```

Comment:
should we move this var to outside of dfs function?

Reply:
Yes, we should do it so that `dr` and `dc` will not be redeclared every time we call `dfs` .
Thanks you @quangh33.